aten/src/ATen/native/Unique.cpp

```diff
@@ -0,0 +1,65 @@
+// Returns unique elements of input tensor.
+
+#include "ATen/ATen.h"
+#include "ATen/Dispatch.h"
+
+#include <set>
+#include <tuple>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace at {
+namespace native{
+
+namespace {
+
+template <template <class...> class set_type, typename scalar_t>
+void _unique_cpu_template(
+    const Tensor& self,
+    const bool return_inverse,
+    Tensor* output,
+    Tensor* inverse_indices) {
+  set_type<scalar_t> set(
```

Comment:
This looks like it assumes that `self` is contiguous. Use Tensor.contiguous() either here or in unique_cpu.

Reply:
Done